spec/rspec/mocks/nil_expectation_warning_spec.rb

```diff
@@ -24,6 +24,13 @@ module Mocks
         nil.foo
       end
 
+      it "raises an error when expectations on nil are disallowed" do
+        disallow_message_expectations_on_nil
+
+        expect { expect(nil).to receive(:foo)     }.to raise_error(StandardError)
```

Comment:
It will eventually be a `RSpec::Mocks::MockExpectationError` if you use the error generator like Myron suggests!